File: R/cdiscbuilder/tests/testthat/test-portable-registry.R
test_that("the shared portable registry is valid", {
  root <- normalizePath(file.path(testthat::test_path(), "../../../.."))
  registry_path <- file.path(root, "yaml/registry/portable-functions.yaml")
  registry <- .load_portable_registry(registry_path)

  expect_identical(registry$core$registry_version, "1.0.0")
  expect_identical(registry$core$namespace, "core")
})

test_that("R passes the shared conformance fixtures", {
  root <- normalizePath(file.path(testthat::test_path(), "../../../.."))
  counts <- .run_portable_conformance(
    file.path(root, "yaml/registry/portable-functions.yaml"),
    file.path(root, "yaml/registry/conformance.yaml")
  )

  expect_gte(unname(counts[["evaluation_cases"]]), 19)
  expect_gte(unname(counts[["validation_cases"]]), 7)
})

File: python/tests/test_portable_registry.py
import unittest
from pathlib import Path

from yamaa.portable_registry import PortableRegistry, run_conformance


ROOT = Path(__file__).resolve().parents[2]
REGISTRY = ROOT / "yaml/registry/portable-functions.yaml"
FIXTURES = ROOT / "yaml/registry/conformance.yaml"


class PortableRegistryTest(unittest.TestCase):
    def test_core_registry_is_valid(self):
        registry = PortableRegistry.load(REGISTRY)
        self.assertEqual(registry.core["registry_version"], "1.0.0")
        self.assertEqual(registry.core["namespace"], "core")

    def test_shared_conformance_fixtures(self):
        evaluation_count, validation_count = run_conformance(REGISTRY, FIXTURES)
        self.assertGreaterEqual(evaluation_count, 19)
        self.assertGreaterEqual(validation_count, 7)


if __name__ == "__main__":
    unittest.main()
